MaxText/layers/linears.py

```diff
@@ -389,7 +395,14 @@ def unpermute(self, intermediate, sorted_selected_experts, weights):
           reshaped_weights.astype(jnp.float32),
           precision=matmul_precision,
       )
-    return output.reshape(-1, self.config.max_target_length, self.config.emb_dim // tensor_parallelism).astype(self.dtype)
+    updated_batch = int(self.config.per_device_batch_size * jax.device_count() // self.config.ici_fsdp_parallelism)
```

Comment:
An assertion may be needed if the value is indivisible cc @mailvijayasingh @ZhiyuLi-goog